### ISSUE TITLE
[BUG]: Python doxygen doc generation 

### ISSUE THREAD

**Issue by martinfalk** [inviwo/inviwo#1309]
### Required prerequisites

- [X] Make sure you've read the [documentation](https://inviwo.org). Your issue may be addressed there.
- [X] Search the [issue tracker](https://github.com/inviwo/inviwo/issues) and [Discussions](https://github.com/inviwo/inviwo/discussionss) to verify that this hasn't already been reported. +1 or comment there if it has.
- [X] Consider asking first in [Slack](https://join.slack.com/t/inviwo/shared_invite/enQtNTc2Nzc2NDQwNzIxLTRiMWM1ZWJiYjljZjkwNWE3OTk3MzYxODZlMDUyMzRmZjUzMzBiZjVhNTM3NWUyNzU1MjI4OWJjMzdkODViMzM) or in a [Discussion](https://github.com/inviwo/inviwo/discussionss).

### Problem description

When enabling `IVW_DOXYGEN_PROJECT`, the build fails for `DOXY-Python` as it expects the source files in the build folder to be `.py` and not `.pyd`.

Same failure occurs in github actions VS build.

### Inviwo version

latest master

### Operating system?

Windows

### Compiler / Version

MSVC 2022

### Inviwo log

```text
1>------ Build started: Project: DOXY-Python, Configuration: RelWithDebInfo x64 ------
1>Generate Inviwo python API documentation with sphinx
1>Running Sphinx v5.0.1
1>making output directory... done
1>[autosummary] generating autosummary for: index.rst
1>
1>CUSTOMBUILD : Extension error (sphinx.ext.autosummary):
1>Handler <function process_generate_options at 0x000001F3EEBA08B0> for event 'builder-inited' threw an exception (exception: source is not a .py file: 'build-vs\\bin\\RelWithDebInfo\\inviwopy.cp310-win_amd64.pyd')
1>C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(245,5): error MSB8066: Custom build for 'build-vs\CMakeFiles\5c7356f6e482fb847eb26ad3dccf416d\DOXY-Python.rule' exited with code 2.
```

**Comment by petersteneteg** [comment]
sphinx should most definitely work on the pyd file not any py files.

**Comment by martinfalk** [comment]
Unsure how to convince `sphinx.ext.autosummary` to accept binary `.pyd` files ... The [docs](https://www.sphinx-doc.org/en/master/usage/extensions/autosummary.html) do not offer any option.

**Comment by ResearchDaniel** [comment]
Should work out of the box?
https://github.com/sphinx-doc/sphinx/issues/5728

**Comment by petersteneteg** [comment]
are all your python versions consistent? if you run sphinx with python 3.9 on a a .pyd for python 3.10. python 3.9 will not find the pyd file at all

**Comment by martinfalk** [comment]
I have only installed version 3.10. And running the command in a console does result in the same error.

@ResearchDaniel: the pyd files are found and the libraries are imported. It is just that autosummary does not expect pyds.

**Comment by martinfalk** [comment]
Might be resolved in latest sphinx version 5.0.2. 
See https://github.com/sphinx-doc/sphinx/issues/10509
